guacamole/src/main/webapp/app/manage/controllers/manageUserController.js

```diff
@@ -164,197 +164,283 @@ angular.module('manage').controller('manageUserController', ['$scope', '$injecto
             return false;
 
         // The administrator can always delete users
-        if (PermissionSet.hasSystemPermission($scope.permissions,
+        if (PermissionSet.hasSystemPermission($scope.permissions[dataSource],
                 PermissionSet.SystemPermissionType.ADMINISTER))
             return true;
 
         // Otherwise, require explicit DELETE permission on the user
-        return PermissionSet.hasUserPermission($scope.permissions,
+        return PermissionSet.hasUserPermission($scope.permissions[dataSource],
             PermissionSet.ObjectPermissionType.DELETE, username);
 
     };
 
     /**
-     * Returns whether the user being edited is read-only, and thus cannot be
-     * modified by the current user.
+     * Returns whether the user being edited within the given data source is
+     * read-only, and thus cannot be modified by the current user.
+     *
+     * @param {String} [dataSource]
+     *     The identifier of the data source to check. If omitted, this will
+     *     default to the currently-selected data source.
      *
      * @returns {Boolean}
      *     true if the user being edited is actually read-only and cannot be
      *     edited at all, false otherwise.
      */
-    $scope.isReadOnly = function isReadOnly() {
-        return !$scope.canSaveUser();
-    };
+    $scope.isReadOnly = function isReadOnly(dataSource) {
 
-    // Pull user attribute schema
-    schemaService.getUserAttributes(dataSource).success(function attributesReceived(attributes) {
-        $scope.attributes = attributes;
-    });
+        // Use currently-selected data source if unspecified
+        dataSource = dataSource || selectedDataSource;
 
-    // Pull user data
-    dataSourceService.apply(userService.getUser, dataSources, username)
-    .then(function usersReceived(users) {
+        // User is read-only if they cannot be saved
+        return !$scope.canSaveUser(dataSource);
 
-        // Get user for currently-selected data source
-        $scope.user = users[dataSource];
+    };
 
-        // Create skeleton user if user does not exist
-        if (!$scope.user) {
-            userExists = false;
-            $scope.user = new User({
-                'username' : username
-            });
-        }
-        else
-            userExists = true;
+    // Update visible account pages whenever available users/permissions changes
+    $scope.$watchGroup(['users', 'permissions'], function updateAccountPages() {
 
         // Generate pages for each applicable data source
         $scope.accountPages = [];
         angular.forEach(dataSources, function addAccountPage(dataSource) {
 
             // Determine whether data source contains this user
-            var linked = dataSource in users;
+            var linked   = $scope.userExists(dataSource);
+            var readOnly = $scope.isReadOnly(dataSource);
+
+            // Account is not relevant if it does not exist and cannot be
+            // created
```

Comment:
Gotta keep lines within 80 chars unless it hurts readability - here's what it looks like in an editor with a big red vertical

![gotta-keep-it-within-80-chars](https://cloud.githubusercontent.com/assets/4632905/9673450/28f0811e-5259-11e5-8bd9-103d7cbf6a3e.png)
 line at 80 columns: